src/wui/progress_data_wrapper.cpp

```diff
@@ -0,0 +1,35 @@
+#include "../../../lib/Marlin/Marlin/src/gcode/lcd/M73_PE.h"
+#include "progress_data_wrapper.h"
+#include <stdio.h>
+
+extern "C" {
+
+bool is_percentage_valid(uint32_t print_dur) {
+    return oProgressData.oPercentDone.mIsActual(print_dur);
```

Comment:
are these thread safe?

Reply:
Shouldn't be a problem. They are as safe as GUI's progress calculations.